sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveHints.scala

```diff
@@ -47,20 +49,51 @@ object ResolveHints {
    *
    * This rule must happen before common table expressions.
    */
-  class ResolveBroadcastHints(conf: SQLConf) extends Rule[LogicalPlan] {
+  class ResolveBroadcastHints(conf: SQLConf, catalog: SessionCatalog) extends Rule[LogicalPlan] {
     private val BROADCAST_HINT_NAMES = Set("BROADCAST", "BROADCASTJOIN", "MAPJOIN")
 
     def resolver: Resolver = conf.resolver
 
-    private def applyBroadcastHint(plan: LogicalPlan, toBroadcast: Set[String]): LogicalPlan = {
+    private def namePartsWithDatabase(nameParts: Seq[String], database: String): Seq[String] = {
+      if (nameParts.size == 1) {
+        database +: nameParts
+      } else {
+        nameParts
+      }
+    }
+
+    private def formatDatabaseName(name: String): String = {
+      if (conf.caseSensitiveAnalysis) name else name.toLowerCase(Locale.ROOT)
+    }
+
+    private def matchedTableIdentifier(
+        nameParts: Seq[String],
+        tableIdent: IdentifierWithDatabase): Boolean = {
+      tableIdent.database match {
+        case Some(db) if catalog.globalTempViewManager.database == formatDatabaseName(db) =>
```

Comment:
Could you use `resolver` here like the following and remove `formatDatabaseName` in line 65~67? Since it's a `SessionCatalog` function, let's avoid duplication. 
```scala
- case Some(db) if catalog.globalTempViewManager.database == formatDatabaseName(db) =>
+ case Some(db) if resolver(catalog.globalTempViewManager.database, db) =>
```

Reply:
Also, we need a case-sensitive test. I made another PR to you for that, https://github.com/maropu/spark/pull/3 .